### ISSUE TITLE
Identify possible IPv4 in IPv6 approaches

### ISSUE THREAD

**Issue by coot** [input-output-hk/ouroboros-network#3677]
* [ ] identify possible IPv4 in IPv6 approaches
* [ ] identify which are being commercially used
* [ ] identify local configuration scenarios - which make sense (i.e does presence of IPv4 negate need for this translation?)
* [ ] identify approach for embedding into code - is this an address translation at "discovery" time an/or an address translation at "use" time?
* [ ] identify testing requirements - including how can this be tested in CI (does a suitable VM testing environment exist? Would one need to be created)

**Comment by njd42** [comment]
One approach is the [NAT64 approach](https://en.wikipedia.org/wiki/IPv6_transition_mechanism) - there is a public address range for this but there is also the ability to run a local service.|
If this approach was the one chosen we would need something like a configuration flag (to enable) along with an option to set the IPv6 prefix (for local services  - such as may be run by large corporates for security reasons)

**Comment by coot** [comment]
Nice, so this can be deployed together with `DNS64`, which allows to get `AAAA` records translated from `A` record.

**Comment by njd42** [comment]
Yes, and that is what an IPv6 only ISP does (though it does have consequences for DNS authentication). However, this does not help with the case where IPv4 addresses are used - hence the potential need for emulating the IPv4 into IPv6 address embedding and the ability to configure that prefix (for non-public networks)

**Comment by gufmar** [comment]
I need to understand if such v4-6 translations are required, important or beneficial in certain cases, or if a certain good amount of nodes operating both a v4 and v6 interface are natural, decentralised and redundant interconnectors for nodes operating only one protocol version.

**Comment by coot** [comment]
@gufmar if there is an ipv6 only node, its pool of known peers will largely consists of unreachable peers (there are very few ipv6 nodes), for security reasons (eclipse attacks) ipv4-6 translation is mandatory.
